박은지 과제 제출합니다

eunjipark123 opened the pull request:
과제 제출합니다! (늦어서 죄송합니다 🥲)

eunjipark123 committed:
과제 제출

과제 제출합니다!

dev-moonduck commented on homework/dags/my_dag.py:
```diff
@@ -1,67 +1,53 @@
+    download_code = BashOperator(
+        task_id='download_code',
+        bash_command=f'curl -k -o {CODE_OUTPUT} '
+                     + f'{CODE_INPUT}'
```
굳이 새로운 라인에 할 필요는 없어보입니다. 혹시 제 예제에서 복사하신거라면 죄송합니다.

eunjipark123 replied:
짧을 때는 안써도 되는데, 생각이 짧았습니다 ㅜㅜ!!

dev-moonduck commented on homework/dags/stock_export.py:
```diff
@@ -0,0 +1,51 @@
+                    """
+                    INSERT INTO stock
+                    VALUES (%s, %s, %s, %s, %s)
+                    """, data)
```
psycopg2의 공식문서에 따르면 executemany는 for문으로 하나하나 execute하는 것보다도 떨어지는 성능이라고 합니다.
https://www.psycopg.org/docs/cursor.html#cursor.executemany
psycopg2를 쓰신다면, 쓰지 말아야할 함수라고 보시면됩니다.

dev-moonduck commented:
전반적으로 크게 흠잡을만한 코드는 없어보입니다(executemany 빼구요 😄 ) 잘하셨어요!

eunjipark123 committed:
과제 수정

1. bash_command 내 불필요한 줄 나눔 삭제
2. executemany -> execute 반복문 사용으로 변경
(다른 접근을 하지 않은 이유 :
- execute_batch()는 psycopg2에서만 쓰이는 것으로 보임
- copy()는 파일을 읽어와서 바로 실행하는 메서드이고, csv에서만 읽어올 수 있는 것으로 보임)

eunjipark123 committed:
execute_batch로 시도

exectue_batch로 시도할 수 있었네요..!

- export 오퍼레이터의 requirements를 psycopg2로 설정
-execute > psycopg2.extra.execute_batch로 변경하여 실행

eunjipark123 committed:
프로젝트 인터프리터 변경

인터프리 문제인 것 같아 수정해봤습니다..;
